src/Adapter/Presenter/Cart/CartPresenter.php

```diff
@@ -585,6 +608,46 @@ private function getTemplateVarVouchers(Cart $cart)
         );
     }
 
+    /**
+     * @param array $cartVoucher
+     *
+     * @return bool
+     */
+    private function cartVoucherHasFreeShipping($cartVoucher)
+    {
+        return !empty($cartVoucher['free_shipping']);
+    }
+
+    /**
+     * @param array $cartVoucher
+     *
+     * @return bool
+     */
+    private function cartVoucherHasPercentReduction($cartVoucher)
+    {
+        return isset($cartVoucher['reduction_percent'])
+            && $cartVoucher['reduction_percent'] > 0
+            && $cartVoucher['reduction_amount'] == '0.00';
+            isset($cartVoucher['reduction_percent'])
```

Comment:
You accept my change without reading it :sweat_smile:

Reply:
damn I read too fast, I thought you only removed the if and put the condition directly in the return...

Reply:
Fixed .